runtime/src/comm/ofi/comm-ofi.c

```diff
@@ -1292,6 +1292,12 @@ struct fi_info* findProvInList(struct fi_info* info,
       DBG_PRINTF_NODE0(DBG_PROV, "skipping T2 interface");
       continue;
     }
+    // libfabric currently doesn't handle FI_SOCKADDR_IB addresses properly.
```

Comment:
Several things:
- Could you move this logic and the T2-related check above into a new address format-specific check routine, defined somewhere near the other detailed provider checking functions like `isGoodCoreProvider()` etc.?
- With regard to the comment, it's not so much libfabric in general as the verbs provider specifically that has trouble with the `FI_SOCKADDR_IB` format addresses, is that correct? If so, rephrase the commentary?
- Finally, only do this for the verbs provider (`isInProvider("verbs", info)`)? I realize this will actually slow the code down since we'll be doing a string comparison and then an integer comparison instead of just the latter, but this only runs at startup and I think the clarity is worthwhile.

Reply:
Please take a look at the changes I made to address your comments.